vessel-runtime/src/main/java/com/textnow/android/vessel/VesselImpl.kt

```diff
@@ -328,12 +517,14 @@ class VesselImpl(
         check(!closeWasCalled) { "Vessel($name:${hashCode()}) was already closed." }
         type.qualifiedName?.let { typeName ->
             return dao.getFlow(typeName)
-                    .distinctUntilChanged()
-                    .map {
-                        it?.data?.let { entity ->
-                            fromJson(entity, type)
-                        }
+                .distinctUntilChanged()
+                .map {
+                    it?.data?.let { entity ->
+                        val data = fromJson(entity, type)
+                        cache?.set(typeName, data as Any)
+                        data
                     }
```

Comment:
Optional, but you could use `also` here in place of the local val for these blocks. Use whatever you prefer for readability though
```suggestion
                    it?.data?.let { entity ->
                        fromJson(entity, type).also { data ->
                            cache?.set(typeName, data as Any)
                        }
                    }
```

Reply:
Thanks!  Good callout, and done!